### PR TITLE
We now identify players based on a ID! Less traffic. Yay.

### PR DESCRIPTION
**Pull request by BeeHiveJava**

### CODE REVIEW

File: Soccar [UI]/src/nl/socnet/message/ChatMessage.java [hunk 1]
```diff
@@ -13,24 +13,24 @@
...
 public final class ChatMessage extends Message {
 
-    private final String username;
+    private final int playerId;
     private final Privilege privilege;
     private final String message;
 
-    public ChatMessage(String username, Privilege privilege, String message) {
-        this.username = username;
+    public ChatMessage(int playerId, Privilege privilege, String message) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/ChatMessage.java:20]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/ChatMessage.java [hunk 2]
```diff
@@ -13,24 +13,24 @@
...
+        this.playerId = playerId;
         this.privilege = privilege;
         this.message = message;
     }
 
     public ChatMessage(String message) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/ChatMessage.java:26]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/soccar/ui/fx/controller/SessionViewFXMLController.java [hunk 3]
```diff
@@ -138,7 +138,7 @@ private void leaveRoom() {
         Room room = currentPlayer.getCurrentSession().getRoom();
         
         TeamColour colour = room.getTeamBlue().getPlayers().stream().filter(currentPlayer::equals).count() > 0 ? TeamColour.BLUE : TeamColour.RED;
```
Comment (TehMessiah) [inline Soccar [UI]/src/nl/soccar/ui/fx/controller/SessionViewFXMLController.java]: MAJOR Remove this unused "colour" local variable. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1481)
MAJOR Remove this useless assignment to local variable "colour". [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1854)

File: Soccar [UI]/src/nl/socnet/message/PlayerJoinedSessionMessage.java [hunk 4]
```diff
@@ -1,39 +1,59 @@
 package nl.socnet.message;
 
 import nl.soccar.library.Player;
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/PlayerJoinedSessionMessage.java:3]: MINOR Remove this unused import 'nl.soccar.library.Player'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: Soccar [UI]/src/nl/socnet/message/handler/ChatMessageHandler.java [hunk 5]
```diff
@@ -3,13 +3,18 @@
 import io.netty.buffer.ByteBuf;
 import javafx.application.Platform;
 import nl.soccar.gamecommuncation.util.ByteBufUtilities;
+import nl.soccar.library.Player;
+import nl.soccar.library.Session;
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/handler/ChatMessageHandler.java:7]: MINOR Remove this unused import 'nl.soccar.library.Session'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 13 issues
* MAJOR 2 major
* MINOR 11 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [SessionViewFXMLController.java#L177](https://github.com/PTS3-S34A/User-Interface/blob/f79e6efad180a9bcd02684258ef5f305e24987ca/Soccar [UI]/src/nl/soccar/ui/fx/controller/SessionViewFXMLController.java#L177): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [PlayerLeaveSessionMessage.java#L3](https://github.com/PTS3-S34A/User-Interface/blob/f79e6efad180a9bcd02684258ef5f305e24987ca/Soccar [UI]/src/nl/socnet/message/PlayerLeaveSessionMessage.java#L3): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessage.java#L3](https://github.com/PTS3-S34A/User-Interface/blob/f79e6efad180a9bcd02684258ef5f305e24987ca/Soccar [UI]/src/nl/socnet/message/PlayerLeftSessionMessage.java#L3): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [MovePlayerMessageHandler.java#L5](https://github.com/PTS3-S34A/User-Interface/blob/f79e6efad180a9bcd02684258ef5f305e24987ca/Soccar [UI]/src/nl/socnet/message/handler/MovePlayerMessageHandler.java#L5): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeaveSessionMessageHandler.java#L4](https://github.com/PTS3-S34A/User-Interface/blob/f79e6efad180a9bcd02684258ef5f305e24987ca/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeaveSessionMessageHandler.java#L4): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessageHandler.java#L8](https://github.com/PTS3-S34A/User-Interface/blob/f79e6efad180a9bcd02684258ef5f305e24987ca/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeftSessionMessageHandler.java#L8): Remove this unused import 'nl.soccar.library.Team'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessageHandler.java#L9](https://github.com/PTS3-S34A/User-Interface/blob/f79e6efad180a9bcd02684258ef5f305e24987ca/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeftSessionMessageHandler.java#L9): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: Soccar [UI]/src/nl/soccar/ui/fx/controller/SessionViewFXMLController.java [hunk 6]
```diff
@@ -136,9 +130,8 @@ private void leaveRoom() {
         ClientController controller = ClientController.getInstance();
         Connection connection = controller.getCurrentConnection();
         Room room = currentPlayer.getCurrentSession().getRoom();
```
Comment (TehMessiah) [inline Soccar [UI]/src/nl/soccar/ui/fx/controller/SessionViewFXMLController.java]: MAJOR Remove this unused "room" local variable. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1481)
MAJOR Remove this useless assignment to local variable "room". [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1854)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 13 issues
* MAJOR 2 major
* MINOR 11 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [SessionViewFXMLController.java#L170](https://github.com/PTS3-S34A/User-Interface/blob/a67752a1a312e558edb343ac8dbf5462017fa7f9/Soccar [UI]/src/nl/soccar/ui/fx/controller/SessionViewFXMLController.java#L170): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [PlayerLeaveSessionMessage.java#L3](https://github.com/PTS3-S34A/User-Interface/blob/a67752a1a312e558edb343ac8dbf5462017fa7f9/Soccar [UI]/src/nl/socnet/message/PlayerLeaveSessionMessage.java#L3): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessage.java#L3](https://github.com/PTS3-S34A/User-Interface/blob/a67752a1a312e558edb343ac8dbf5462017fa7f9/Soccar [UI]/src/nl/socnet/message/PlayerLeftSessionMessage.java#L3): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [MovePlayerMessageHandler.java#L5](https://github.com/PTS3-S34A/User-Interface/blob/a67752a1a312e558edb343ac8dbf5462017fa7f9/Soccar [UI]/src/nl/socnet/message/handler/MovePlayerMessageHandler.java#L5): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeaveSessionMessageHandler.java#L4](https://github.com/PTS3-S34A/User-Interface/blob/a67752a1a312e558edb343ac8dbf5462017fa7f9/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeaveSessionMessageHandler.java#L4): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessageHandler.java#L8](https://github.com/PTS3-S34A/User-Interface/blob/a67752a1a312e558edb343ac8dbf5462017fa7f9/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeftSessionMessageHandler.java#L8): Remove this unused import 'nl.soccar.library.Team'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessageHandler.java#L9](https://github.com/PTS3-S34A/User-Interface/blob/a67752a1a312e558edb343ac8dbf5462017fa7f9/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeftSessionMessageHandler.java#L9): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Review by Ranray — APPROVED** [review]

**Comment by ghost** [comment]
SonarQube analysis reported 11 issues
* MINOR 11 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [SessionViewFXMLController.java#L169](https://github.com/PTS3-S34A/User-Interface/blob/0b0159259e74e1684a66076b9e948568c9146425/Soccar [UI]/src/nl/soccar/ui/fx/controller/SessionViewFXMLController.java#L169): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [PlayerLeaveSessionMessage.java#L3](https://github.com/PTS3-S34A/User-Interface/blob/0b0159259e74e1684a66076b9e948568c9146425/Soccar [UI]/src/nl/socnet/message/PlayerLeaveSessionMessage.java#L3): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessage.java#L3](https://github.com/PTS3-S34A/User-Interface/blob/0b0159259e74e1684a66076b9e948568c9146425/Soccar [UI]/src/nl/socnet/message/PlayerLeftSessionMessage.java#L3): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [MovePlayerMessageHandler.java#L5](https://github.com/PTS3-S34A/User-Interface/blob/0b0159259e74e1684a66076b9e948568c9146425/Soccar [UI]/src/nl/socnet/message/handler/MovePlayerMessageHandler.java#L5): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeaveSessionMessageHandler.java#L4](https://github.com/PTS3-S34A/User-Interface/blob/0b0159259e74e1684a66076b9e948568c9146425/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeaveSessionMessageHandler.java#L4): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessageHandler.java#L8](https://github.com/PTS3-S34A/User-Interface/blob/0b0159259e74e1684a66076b9e948568c9146425/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeftSessionMessageHandler.java#L8): Remove this unused import 'nl.soccar.library.Team'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessageHandler.java#L9](https://github.com/PTS3-S34A/User-Interface/blob/0b0159259e74e1684a66076b9e948568c9146425/Soccar [UI]/src/nl/socnet/message/handler/PlayerLeftSessionMessageHandler.java#L9): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Pull request merged**